pkg/kubectl/cmd/set/set_subject.go

```diff
@@ -219,7 +219,6 @@ func (o *SubjectOptions) Run(f cmdutil.Factory, fn updateSubjects) error {
 
 		transformed, err := updateSubjectForObject(info.Object, subjects, fn)
 		if transformed && err == nil {
-			// TODO: switch UpdatePodSpecForObject to work on v1.PodSpec, use info.VersionedObject, and avoid conversion completely
 			versionedEncoder := legacyscheme.Codecs.EncoderForVersion(o.Encoder, info.Mapping.GroupVersionKind.GroupVersion())
```

Comment:
should be able to remove `versionedEncoder ...` and use o.Encoder directly

Reply:
`set_subject` and `set_selector` not use f.UpdatePodSpecForObject, will fix it when other test pass ci.

Reply:
when i use v1beta2.StatefulSet yaml will occur:
`
statefulsets/minio converting (v1beta2.StatefulSet).ObservedGeneration to (v1beta1.StatefulSet).ObservedGeneration: type names don't match (int64, ), and no conversion 'func (int64, *int64) error' registered.`

Reply:
this means you're trying to convert between two external versions directly and falling back to reflective conversion paths, which is very odd.

Reply:
`testapi.Default = testapi.Groups[input.apiGroup]` in the test is using the default test version for that group. I'm guessing that does not match the version of the API object in the test fixture

Reply:
when i use binary alse reproduce this:
```admindeMacBook-Pro-2:kubectl admin$ ./kubectl set image -f ../../examples/storage/minio/minio-distributed-statefulset.yaml *=gninxxx --local -o yaml
onverting (v1beta2.StatefulSet).ObservedGeneration to (v1beta1.StatefulSet).ObservedGeneration: type names don't match (int64, ), and no conversion 'func (int64, *int64) error' registered.```